TrafficReplayer/src/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -56,57 +60,106 @@ public TrafficReplayer(URI serverUri)
         packetHandlerFactory = new NettyPacketToHttpHandlerFactory(serverUri);
     }
 
-    private static void exitWithUsageAndCode(int statusCode) {
-        System.err.println("Usage: <log directory> <destination_url>");
-        System.exit(statusCode);
+
+    static class Parameters {
```

Comment:
I do not understand the java syntax going on here at all, but this is definitely an improvement. Does having a "name" field mean that the name needs to be provided? (but the URI can be inferred as the first parameter?)